src/components/NcRichText/NcRichText.vue

```diff
@@ -449,24 +471,37 @@ export default {
 
 						if (!tag.startsWith('#')) {
 							if (this.useExtendedMarkdown) {
+								let nestedList
 								if (tag === 'li' && Array.isArray(children)
 									&& children[0].tag === 'input'
 									&& children[0].data.attrs.type === 'checkbox') {
-									const [inputNode, , label] = children
-									const id = 'markdown-input-' + GenRandomId(5)
+									const [inputNode, ...labelParts] = children
+
+									const nestedListIndex = labelParts.findIndex((child) => child.tag === 'ul' || child.tag === 'li')
+									if (nestedListIndex !== -1) {
+										nestedList = labelParts[nestedListIndex]
+										if (nestedList.data?.attrs) {
+											nestedList.data.attrs.style = 'margin-inline-start: 20px;'
+										}
+										labelParts.splice(nestedListIndex)
+									}
+
+									const id = this.parentId + '-markdown-input-' + GenRandomId(5)
 									const inputComponent = h(NcCheckboxRadioSwitch, {
 										attrs: {
 											...inputNode.data.attrs,
 											id,
 											disabled: !this.interactive,
 										},
 										on: {
-											'update:checked': (value) => {
-												this.$emit('interact:todo', { id, label, value })
+											'update:checked': () => {
+												this.$emit('interact:todo', id)
```

Comment:
Formally, this is a breaking change. Public event payload has changed.

Though it could be fine, it is not a widely used feature.